### DISCUSSION TITLE
OAuth2AuthorizationCodeBearer: use id_token instead access_token

### DISCUSSION THREAD

**Discussion by datasatanic** [fastapi/fastapi#8557] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
app: FastAPI = FastAPI(title="pyauth", debug=True,
                       swagger_ui_init_oauth={"clientId": "eN2PZiKgIwt3UZEDHtttivsUJdNV6W8y",
                                              "clientSecret": "hvo_secret_5OYAbYzfYi9Oh9dKvPqiBjwI4jT5lT4AhZwezm0Gi7HFYCMPpqoladkAWRcDG2Ci",
                                              "appName": "pyauth",
                                              "scopes": "openid",
                                              "useBasicAuthenticationWithAccessCodeGrant": True,
                                              },
                       )

oauth = OAuth2AuthorizationCodeBearer("http://127.0.0.1:8200/ui/vault/identity/oidc/provider/test/authorize",
                                      "http://127.0.0.1:8200/v1/identity/oidc/provider/test/token",
                                      scopes={"admins": "asda", "user": "users"},
                                      scheme_name="Vault"
                                      )
```

### Description

I use OAuth2AuthorizationCodeBearer with external Oidc provider (Vault OIDC Provider).  Identity server receive acess_token and id_token. When calling my api it send access_token, but I need id_token. is it possible to send id_token instead?

In swagger [issue](https://github.com/swagger-api/swagger-ui/issues/4084) I found info about x-tokenName, but I can't understand how I can use it in FastApi. 

### Operating System

macOS

### FastAPI Version

0.71

### Python Version

3.9

**Comment by leonardodepaula** [comment]
The object's key that the API returns is not defined by the OAuth2AuthorizationCodeBearer, but by the endpoint. Check line 38: https://github.com/tiangolo/full-stack-fastapi-postgresql/blob/master/%7B%7Bcookiecutter.project_slug%7D%7D/backend/app/app/api/api_v1/endpoints/login.py

**Comment by cblackst13** [comment]
You can [extend the OpenAPI schema](https://fastapi.tiangolo.com/advanced/extending-openapi/) in FastAPI to specify `x-tokenName` and set it as `id_token`.

```python
from fastapi import FastAPI
from fastapi.openapi.utils import get_openapi

app = FastAPI()

def custom_openapi():
    if app.openapi_schema:
        return app.openapi_schema
    openapi_schema = get_openapi(
        title="OIDC Authenticated API",
        version="1.0.0",
        description="An API that uses OIDC id_token to authenticate",
        routes=app.routes,
    )
    openapi_schema["components"]["securitySchemes"][f'{SCHEMA_NAME}']["x-tokenName"] = "id_token"
    app.openapi_schema = openapi_schema
    return app.openapi_schema

app.openapi = custom_openapi
```

It would be great if this was supported through the OAuth2 modules though.
